File: tiny-event-sourcing-spring-app/src/main/kotlin/ru/quipy/shop/payment/events/PaymentChangeStatusEvent.java
package ru.quipy.shop.payment.events;

import lombok.Getter;
import ru.quipy.core.annotations.DomainEvent;
import ru.quipy.domain.Event;
import ru.quipy.shop.payment.PaymentAggregate;
import ru.quipy.shop.payment.entities.PaymentStatus;

import java.util.UUID;

@DomainEvent(name = "PAYMENT_CHANGE_STATUS_EVENT")
public class PaymentChangeStatusEvent extends Event<PaymentAggregate> {

    @Getter
    private UUID paymentId;

    @Getter
    private PaymentStatus status;

    public PaymentChangeStatusEvent(UUID paymentId, PaymentStatus status) {
        super();
        this.paymentId = paymentId;
        this.status = status;
    }
}

File: tiny-event-sourcing-spring-app/src/main/kotlin/ru/quipy/shop/payment/events/PaymentChangeStatusEvent.kt
package ru.quipy.shop.payment.events

import lombok.Getter
import ru.quipy.core.annotations.DomainEvent
import ru.quipy.domain.Event
import ru.quipy.shop.payment.PaymentAggregate
import ru.quipy.shop.payment.entities.PaymentStatus

import java.util.UUID

@DomainEvent(name = "PAYMENT_CHANGE_STATUS_EVENT")
class PaymentChangeStatusEvent(
    @Getter
    val paymentId: UUID,
    @Getter
    val status: PaymentStatus
) : Event<PaymentAggregate>(
    name = "PAYMENT_CHANGE_STATUS_EVENT"
)